l10n_ch_scan_bvr/wizard/scan_bvr.py

```diff
@@ -485,61 +348,64 @@ def validate_bvr_string(self, cr, uid, ids, context):
         # Explode and check  the BVR Number and structurate it
         #
         data = {}
-        data['bvr_struct'] = self._get_bvr_structurated(
-            bvr_data.bvr_string)
+        data['bvr_struct'] = self._get_bvr_structurated(self.bvr_string)
+        partner_bank_model = self.env['res.partner.bank']
+        partner_bank = False
         # We will now search the account linked with this BVR
         if data['bvr_struct']['domain'] == 'name':
             domain = [('acc_number', '=', data['bvr_struct']['beneficiaire'])]
-            partner_bank_search = self.pool.get('res.partner.bank').search(
-                cr,
-                uid,
-                domain,
-                context=context
-            )
+            partners_bank = partner_bank_model.search(domain)
+            partner_bank = len(partners_bank) == 1 and partners_bank[0]
```

Comment:
`ResPartnerBank` has a constraint of unicity on acc_number thus we can do 

``` python
partner_bank = partner_bank_model.search(domain, limit=1)
```

And in case it returns nothing `partner_bank` will simply be an empty record set